File: prefix_sum/ruby/04-04_string_count_1.py
# 【区間内の個数 (文字列) 】 区間内の個数 (文字列) 4 (paizaランク C 相当)
# https://paiza.jp/works/mondai/prefix_sum_problems/prefix_sum_problems__string_count_boss

INPUT1 = """\
10 3 8
bwwbwbbwbw
"""
OUTPUT1 = """\
3
"""

INPUT2 = """\
10 3 8
wbbwbbbwwb
"""
OUTPUT2 = """\
4
"""


def solve(input_str):
    # 入力
    input_lines = input_str.split("\n")
    _, x, y = map(int, input_lines[0].split())
    str = input_lines[1]

    # b: 1, w: 0
    b = [1 if c == "b" else 0 for c in list(str)]

    # 累積和
    s = [0]
    for num in b:
        s.append(s[-1] + num)

    # 区間の "b" の数
    return s[y] - s[x - 1]


print(solve(open(0).read()))

File: prefix_sum/ruby/04-04_string_count_1.rb
# 【区間内の個数 (文字列) 】 区間内の個数 (文字列) 4 (paizaランク C 相当)
# https://paiza.jp/works/mondai/prefix_sum_problems/prefix_sum_problems__string_count_boss

INPUT1 = <<"EOS"
10 3 8
bwwbwbbwbw
EOS
OUTPUT1 = <<"EOS"
3
EOS

INPUT2 = <<"EOS"
10 3 8
wbbwbbbwwb
EOS
OUTPUT2 = <<"EOS"
4
EOS

def solve(input_str)
  input_lines = input_str.split("\n")
  _, x, y = input_lines[0].split.map(&:to_i)
  str = input_lines[1]

  # b: 1, w: 0
  b = str.chars.map { |c| c == "b" ? 1 : 0 }

  # 累積和
  s = [0]
  b.each { |num| s << s[-1] + num }

  # 区間の "b" の数
  s[y] - s[x - 1]
end

puts solve($stdin.read)

exit
# 入力
_, x, y = gets.split.map(&:to_i)
str = gets.chomp

# b: 1, w: 0
b = str.chars.map { |c| c == "b" ? 1 : 0 }

# 累積和
s = [0]
b.each { |num| s << s[-1] + num }

# 区間の "b" の数
num_of_b = s[y] - s[x - 1]
puts num_of_b
